components/eam/src/chemistry/modal_aero/modal_aero_coag.F90

```diff
@@ -1,2337 +1,1363 @@
-              + ybetaij0(6)*qnum_tmpc(npca) + ybetaij0(8)*qnum_tmpc(nmacc)
-         tmpa = max( 0.0_r8, deltat*tmpa )
-         tmpb = max( 0.0_r8, deltat*ybetaii0(5) )
-         tmpn = qnum_tmpa(nmait)
-         if (tmpa < 1.0e-5_r8) then
-            qnum_tmpb(nmait) = tmpn / &
-               ( 1.0_r8 + (tmpa+tmpb*tmpn)*(1.0_r8 + 0.5_r8*tmpa) )
-         else
-            tmpc = exp(-tmpa)
-            qnum_tmpb(nmait) = tmpn*tmpc / &
-               ( 1.0_r8 + (tmpb*tmpn/tmpa)*(1.0_r8-tmpc) )
-         end if
-         qnum_tmpc(nmait) = (qnum_tmpa(nmait) + qnum_tmpb(nmait))*0.5_r8
-      end if
+    ! Loss to accumulation mode, harmonic mean
 
+    qn12 = coagnc0 * coagfm0 / ( coagnc0 + coagfm0 )
 
-! now calculate mass transfers between modes
-! the transfer amounts are calculated using as an exponential decay of
-!     the initial mass concentrations, 
-!     where the decay rate is calculated using the average (over time step) 
-!     number concentrations for each mode
-! the mass transfer calculations are first-order accurate in time, 
-!     because the mass transferred out of a mode does not
-!     include any mass transferred in during the time step
-! with this approach, the ordering is not important, but the mass transfer
-!     calculations are done in the reverse order of the number loss calculations
-
-! mass transfer out of marine-organics aitken mode
-!    uses average number conc. for accum, aitken, pcarbon, and marine-org accum modes
-      if (nmait > 0) then
-         tmp1 = max( 0.0_r8, ybetaij3(5)*qnum_tmpc(nacc) )
-         tmp2 = max( 0.0_r8, ybetaij3(6)*qnum_tmpc(npca) )
-         tmp3 = max( 0.0_r8, ybetaij3(7)*qnum_tmpc(nait) )
-         tmp4 = max( 0.0_r8, ybetaij3(8)*qnum_tmpc(nmacc) )
-         tmpa = tmp1 + tmp2 + tmp3 + tmp4
-         tmpc = deltat*tmpa
-         if (tmpc > epsilonx2) then
-            ! calc coag change only when it is not ~= zero
-            tmp_xf = 1.0_r8 - exp(-tmpc)
-            tmp2 = tmp2/tmpa
-            tmp3 = tmp3/tmpa
-            tmp4 = tmp4/tmpa
-            tmp1 = 1.0_r8 - (tmp2 + tmp3 + tmp4)
-            do iaer = 1, naer
-               tmp_dq = tmp_xf*qaer_tmpa(iaer,nmait)
-               qaer_tmpb(iaer,nmait) = qaer_tmpb(iaer,nmait) - tmp_dq
-               qaer_tmpb(iaer,nacc ) = qaer_tmpb(iaer,nacc ) + tmp_dq*tmp1
-               qaer_tmpb(iaer,npca ) = qaer_tmpb(iaer,npca ) + tmp_dq*tmp2
-               qaer_tmpb(iaer,nait ) = qaer_tmpb(iaer,nait ) + tmp_dq*tmp3
-               qaer_tmpb(iaer,nmacc) = qaer_tmpb(iaer,nmacc) + tmp_dq*tmp4
-               qaer_del_coag_in(iaer,i_agepair_pca ) &
-                                    = qaer_del_coag_in(iaer,i_agepair_pca ) + tmp_dq*tmp2
-               qaer_del_coag_in(iaer,i_agepair_macc) &
-                                    = qaer_del_coag_in(iaer,i_agepair_macc) + tmp_dq*tmp4
-            end do
-         end if
-      end if
+    !-----------------------------------------------------------------
```

Comment:
Same here, a new subroutine for third moment.

Reply:
Done.